src/Namotion.Interceptor.Mqtt/Client/MqttSubjectClientSource.cs

```diff
@@ -0,0 +1,378 @@
+
+        _isStarted = true;
+
+        return new MqttConnectionLifetime(async () =>
+        {
+            if (_client?.IsConnected == true)
+            {
+                await _client.DisconnectAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+        });
+    }
+
+    /// <inheritdoc />
+    public Task<Action?> LoadInitialStateAsync(CancellationToken cancellationToken)
+    {
+        // Retained messages are received through the normal message handler: No separate initial load needed/possible
+        return Task.FromResult<Action?>(null);
+    }
+
+    /// <inheritdoc />
+    public async ValueTask WriteChangesAsync(ReadOnlyMemory<SubjectPropertyChange> changes, CancellationToken cancellationToken)
+    {
+        var client = _client;
+        if (client is null || !client.IsConnected)
+        {
+            throw new InvalidOperationException("MQTT client is not connected.");
+        }
+
+        var length = changes.Length;
+        if (length == 0) return;
+
+        // Rent array from pool for messages
+        var messagesPool = ArrayPool<MqttApplicationMessage>.Shared;
+        var messages = messagesPool.Rent(length);
+        var messageCount = 0;
+        try
+        {
+            var changesSpan = changes.Span;
+
+            // Build all messages first
+            for (var i = 0; i < length; i++)
+            {
+                var change = changesSpan[i];
+                var property = change.Property.TryGetRegisteredProperty();
+                if (property is null || property.HasChildSubjects)
+                {
+                    continue;
+                }
+
+                var topic = GetCachedTopic(change.Property, property);
+                if (topic is null) continue;
+
+                byte[] payload;
+                try
+                {
+                    payload = _configuration.ValueConverter.Serialize(
+                        change.GetNewValue<object?>(),
+                        property.Type);
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "Failed to serialize value for property {PropertyName}.", property.Name);
+                    continue;
+                }
+
+                var message = new MqttApplicationMessage
+                {
+                    Topic = topic,
+                    PayloadSegment = new ArraySegment<byte>(payload),
+                    QualityOfServiceLevel = _configuration.DefaultQualityOfService,
+                    Retain = _configuration.UseRetainedMessages
+                };
+
+                if (_configuration.SourceTimestampPropertyName is not null)
+                {
+                    message.UserProperties =
+                    [
+                        new MqttUserProperty(
+                            _configuration.SourceTimestampPropertyName,
+                            change.ChangedTimestamp.ToUnixTimeMilliseconds().ToString())
```

Comment:
add a "convert" method to config with default implementation like this? same for server side